Fix -Wimplicit-function-declaration error with gcc 14.

```
In file included from apk.c:12:
/usr/include/apk/apk_database.h: In function 'apk_name_cmp_display':
/usr/include/apk/apk_database.h:216:16: error: implicit declaration of function
'strcasecmp'; did you mean 'strncmp'? [-Wimplicit-function-declaration]
  216 |         return strcasecmp(a->name, b->name) ?: strcmp(a->name, b->name);
      |                ^~~~~~~~~~
      |                strncmp
```

--- collectd-apk-0.2.0-origin/apk.c
+++ collectd-apk-0.2.0/apk.c
@@ -1,6 +1,8 @@
 // SPDX-FileCopyrightText: 2022-present Jakub Jirutka <jakub@jirutka.cz>
 // SPDX-License-Identifier: GPL-2.0-or-later
 
+#define _DEFAULT_SOURCE
+
 #include <assert.h>
 #include <errno.h>
 #include <stddef.h>
